### PR TITLE
feature: menu redesign M3

### PR DESCRIPTION
**Pull request by sergeysenja1992**
- add support for M3 menu configuration
> - add mat-sidenav
> - add categories component
> - add menu service
> - updated image logo component
> - updated sidebar component
> - updated navbar toggle widget component
> - add theming
> - add menu animations
- refactored activeItemPathPatterns

### CODE REVIEW

File: packages/components/menu/menu-categories/menu-categories.component.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import {MatIconModule} from '@angular/material/icon';
+import {XmTranslationModule} from '@xm-ngx/translation';
+import {BrandLogo, MenuCategory} from '../menu.interface';
+import {MenuService} from '../menu.service';
+import {Observable} from 'rxjs';
+import {takeUntilOnDestroy, takeUntilOnDestroyDestroy} from '@xm-ngx/operators';
+import {MatButtonModule} from '@angular/material/button';
+import {Router, RouterLink, RouterLinkActive} from '@angular/router';
+import {hideCategories} from '../menu.animation';
+import {MenuCategoriesClassesEnum} from '../menu.model';
```
Comment (sergeysenja1992) [inline packages/components/menu/menu-categories/menu-categories.component.ts:12]: setup formatter. add spaces

File: packages/components/menu/menu.component.ts [hunk 2]
```diff
@@ -145,57 +193,117 @@ export class MenuComponent implements OnInit, OnDestroy {
...
+            if (this.previousActiveNode) {
+                this.previousActiveNode.isActiveRoute = false;
+            }
+            activeNode.isActiveRoute = true;
+            this.previousActiveNode = activeNode;
+        }
+    }
+
+    public observeSectionsFiltering(): void {
+        this.menuService.hoveredCategory
```
Comment (sergeysenja1992) [inline packages/components/menu/menu.component.ts:238]: too complex method split to methods

File: src/app/layouts/layout.module.ts [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
-import { XmInputPatternModule } from '@xm-ngx/components/inputPattern';
-import { XmTimelineModule } from '@xm-ngx/timeline/timeline-widget';
-import { XmFooterComponent } from '@xm-ngx/components/footer';
-import { XmMainComponent } from './main.component';
-import { XmNavbarModule } from '@xm-ngx/administration/navbar';
-import { PageRibbonComponent } from '@xm-ngx/components/page-ribbon';
-import { JhiAlertErrorComponent } from '@xm-ngx/error-messages';
-import { CommonModule } from '@angular/common';
-import { XmHeatmapContainerComponent } from '@xm-ngx/components/navbar-heatmap-widget';
+import {NgModule} from '@angular/core';
```
Comment (sergeysenja1992) [inline src/app/layouts/layout.module.ts:1]: =(

File: packages/components/menu/menu-categories/menu-categories.component.ts (hunk 1 above)
Comment (denisogolubev) [reply packages/components/menu/menu-categories/menu-categories.component.ts]: fixed

File: packages/components/menu/menu.component.ts (hunk 2 above)
Comment (denisogolubev) [reply packages/components/menu/menu.component.ts:236]: done

File: src/app/layouts/layout.module.ts (hunk 3 above)
Comment (denisogolubev) [reply src/app/layouts/layout.module.ts]: fixed

**Pull request closed without merging**